src/gui/mainwindow.cpp

```diff
@@ -1366,19 +1364,73 @@ void MainWindow::on_actionRSS_Reader_triggered()
 
 void MainWindow::on_actionSearch_engine_triggered()
 {
-#ifdef Q_OS_WIN
     if (!has_python && actionSearch_engine->isChecked()) {
-        bool res = false;
+        int pythonVersion = Utils::Misc::pythonVersion();
 
         // Check if python is already in PATH
-        if (Utils::Misc::pythonVersion() > 0)
-            res = true;
-        else
-            res = addPythonPathToEnv();
+        if (pythonVersion > 0)
+            Logger::instance()->addMessage(tr("Python found in %1").arg("PATH"), Log::INFO); // Prevent translators from messing with PATH
+#ifdef Q_OS_WIN
+        else if (addPythonPathToEnv())
+            pythonVersion = Utils::Misc::pythonVersion();
+#endif
+
+        bool res = false;
+
+        if (pythonVersion == 2) {
+            // Check if python 2.7.x or later
+            QString version = Utils::Misc::pythonVersionComplete().trimmed();
+            QStringList splitted = version.split('.');
+            if (splitted.size() > 1) {
+                int middleVer = splitted.at(1).toInt();
+                if (middleVer < 7) {
+                    QMessageBox::information(this, tr("Old Python Interpreter"), tr("Your Python version is %1, which is too old. You need at least version 2.7.0 for python2 or 3.3.0 for python3.").arg(version));
+                    actionSearch_engine->setChecked(false);
+                    Preferences::instance()->setSearchEnabled(false);
+                    return;
+                }
+                else {
+                    res = true;
+                }
+            }
+            else {
+                QMessageBox::information(this, tr("Undetermined Python version"), tr("Couldn't decode your Python version: %1").arg(version));
+                actionSearch_engine->setChecked(false);
+                Preferences::instance()->setSearchEnabled(false);
+                return;
+            }
+        }
+        else if (pythonVersion == 3) {
+            // Check if python 3.3.x or later
+            QString version = Utils::Misc::pythonVersionComplete().trimmed();
+            QStringList splitted = version.split('.');
+            if (splitted.size() > 1) {
+                int middleVer = splitted.at(1).toInt();
+                if (middleVer < 3) {
+                    QMessageBox::information(this, tr("Old Python Interpreter"), tr("Your Python version is %1, which is too old. You need at least version 2.7.0 for python2 or 3.3.0 for python3.").arg(version));
```

Comment:
Your Python version (%1) is too old. You need at least version 2.7.0 for python2 or 3.3.0 for python3. Search engine disabled.

Maybe recommend the user to go for 3.x...